common_utils.sh

```
@@ -270,22 +270,80 @@ function bdist_wheel_cmd {
     cp dist/*.whl $abs_wheelhouse
 }
 
+function run_command_universal2 {
+    if [[ "${PLAT:-}" == "universal2" ]]; then
```

Could do with some comments here.

Any chance of a refactor so this reads better - maybe something like:

```
if [[ "${PLAT:-}" == "universal2" ]]; then
    (macos_intel_build_wrap $@)
fi
if [[ "${PLAT:-}" == "universal2" || "${PLAT:-}" == "arm64" ]]; then
    (macos_arm64_build_wrap $@)
fi
 if [[ "${PLAT:-}" == "universal2" ]]; then
    fuse_macos_intel_arm64
fi
```

Also - I wonder if we can put all this logic into `osx_steps.sh`, overwriting a default `wrap_wheel_builder` - so we can hide this complexity from non-Mac people?